### PR TITLE
[DSCP-194] Consider mempool everywhere

### PR DESCRIPTION
**Pull request by Martoon-00**
I advice to review on per-commit basis, each commit has own description.
Also, I wish all the code here to be taken critically (especially in cosmetics), I don't know what I was doing.

Tested:
- [x] Wallet shows "pending" expenses
* [x] It shows them as pending temporally
* [x] Several payments from same account can be made withing just one slot.

### CODE REVIEW

**Comment by Martoon-00** [comment]
Could you elaborate on "aligning code in columns"?

**Comment by goodsoft** [comment]
I mean this: 
image

**Comment by Martoon-00** [comment]
Meh, I see. `stylish-haskell` does it automatically for me according to config file in repo.

**Comment by Martoon-00** [comment]
AFAUI and if I don't mix anything up, this happens because wallet can't communicate with witnesses on cluster.

**Comment by goodsoft** [comment]
I think it might be because the wallet in this PR is not compatible with the deployed cluster.
You should run your own witness node from this PR.

**Comment by goodsoft** [comment]
Actually, @Martoon-00, please update `explorer.yaml` spec with regards to your changes.

**Comment by Martoon-00** [comment]
Right, thanks :+1:

File: witness/src/Dscp/Witness/SDLock.hs [hunk 1]
```diff
@@ -1,63 +1,80 @@
...
+import Loot.Log (MonadLogging)
 import qualified UnliftIO
 
-import Loot.Base.HasLens (HasLens (..), HasLens')
+import Dscp.Util.TimeLimit (logWarningWaitInf)
 
+-- | Lock for SD operations.
 newtype SDLock = SDLock RawLock.RWLock
 
+data ReadSDLockTag = ReadSDLockTag
```
Comment (volhovm) [inline witness/src/Dscp/Witness/SDLock.hs]: Oh my god, this is actually a very good idea. cc @nasportivnoy

**Comment by Martoon-00** [comment]
Thanks, fixed them.

**Pull request merged**